File: ios/Classes/NeumorphicKitPlugin.m
// filepath: /C:/Users/Muralidharan K/AndroidStudioProjects/neumorphic_kit/ios/Classes/NeumorphicKitPlugin.m
#import "NeumorphicKitPlugin.h"
#import <Flutter/Flutter.h>

@implementation NeumorphicKitPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"neumorphic_kit"
            binaryMessenger:[registrar messenger]];
  NeumorphicKitPlugin* instance = [[NeumorphicKitPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"getPlatformVersion" isEqualToString:call.method]) {
    result([@"iOS " stringByAppendingString:[[UIDevice currentDevice] systemVersion]]);
  } else {
    result(FlutterMethodNotImplemented);
  }
}

@end

File: ios/Classes/NeumorphicKitPlugin.swift
import Flutter
import UIKit

public class NeumorphicKitPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "neumorphic_kit", binaryMessenger: registrar.messenger())
    let instance = NeumorphicKitPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("iOS " + UIDevice.current.systemVersion)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}